### PR TITLE
Add gRPC replication stream

### PR DESCRIPTION
**Pull request by wxing1292**
**What changed?**

**Why?**

**How did you test it?**

**Potential risks**

**Is hotfix candidate?**

### CODE REVIEW

**Review by meiliang86 — APPROVED** [review]

File: client/admin/metric_client.go [hunk 1]
```diff
@@ -90,3 +91,16 @@ func (c *metricClient) finishMetricsRecording(
 	}
 	metricsHandler.Timer(metrics.ClientLatency.GetMetricName()).Record(time.Since(startTime))
 }
+
+func (c *metricClient) StreamReplicationMessages(
+	ctx context.Context,
+	opts ...grpc.CallOption,
+) (_ adminservice.AdminService_StreamReplicationMessagesClient, retError error) {
+
+	metricsHandler, startTime := c.startMetricsRecording(ctx, metrics.AdminStreamReplicationMessagesScope)
```
[the remark this answers is not in the document]
Comment (wxing1292) [reply client/admin/metric_client.go:100]: this is per stream initialization

File: client/admin/client.go [hunk 2]
```diff
@@ -73,3 +75,12 @@ func (c *clientImpl) createContextWithLargeTimeout(parent context.Context) (cont
 	}
 	return context.WithTimeout(parent, c.largeTimeout)
 }
+
+func (c *clientImpl) StreamReplicationMessages(
```
[the remark this answers is not in the document]
Comment (wxing1292) [reply client/admin/client.go:79]: sure, i can change the API name

File: cmd/tools/rpcwrappers/main.go [hunk 3]
```diff
@@ -83,6 +83,14 @@ var (
 		"client.admin.GetReplicationMessages": true,
 	}
 	ignoreMethod = map[string]bool{
+		// stream APIs are not supported. do not generate.
```
[the remark this answers is not in the document]
Comment (yux0) [reply cmd/tools/rpcwrappers/main.go:86]: Can we add a TODO to support auto generate stream APIs?

File: service/history/handler.go [hunk 4]
```diff
@@ -1846,6 +1849,38 @@ func (h *Handler) UpdateWorkflowExecution(
...
+
+	ctxMetadata, ok := metadata.FromIncomingContext(server.Context())
+	if !ok {
+		return serviceerror.NewInvalidArgument("missing cluster & shard ID metadata")
+	}
+	sourceClusterShardID, targetClusterShardID, err := history.DecodeClusterShardMD(ctxMetadata)
+	if err != nil {
+		return err
+	}
+	if targetClusterShardID.ClusterName != h.clusterMetadata.GetCurrentClusterName() {
```
Comment (yux0) [inline service/history/handler.go:1870]: sourceClusterShardID.ClusterName?

File: service/history/ndc/stream.go [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package ndc
```
Comment (yux0) [inline service/history/ndc/stream.go:25]: This should be in replication package?

File: service/history/handler.go (hunk 4 above)
Comment (wxing1292) [reply service/history/handler.go:1870]: this is checking if target cluster is local cluster

File: service/history/ndc/stream.go (hunk 5 above)
Comment (wxing1292) [reply service/history/ndc/stream.go:25]: this package will be moved in later PRs

**Pull request merged**